code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/ManualCarouselNotificationBuilder.kt

```diff
@@ -215,13 +213,14 @@ internal object ManualCarouselNotificationBuilder {
                     )
                 } else {
                     Triple(
+                        imageUris.size - 1,
                         PushTemplateConstants.DefaultValues.MANUAL_CAROUSEL_START_INDEX,
-                        PushTemplateConstants.DefaultValues.MANUAL_CAROUSEL_START_INDEX + 1,
-                        PushTemplateConstants.DefaultValues.MANUAL_CAROUSEL_START_INDEX + 2
+                        PushTemplateConstants.DefaultValues.MANUAL_CAROUSEL_START_INDEX + 1
                     )
                 }
         }
 
+        pushTemplate.centerImageIndex = carouselIndices.second
```

Comment:
Can you elaborate on why this is required?
This method ideally should only responsible for calculating the indices. Manipulating the template here will be difficult to maintain later on. Can this operation be made at a different place?

Reply:
yes, i can move this out of the indices calculation.